File: C/Algorithms/1796.c
int secondHighest(char *s) {
  int max = 47, second_max = 47;
  while (*s != NULL) {
    int c = *s++;
    if (c > 47 && c < 58) {
      if (c > max) {
        if (max > second_max)
          second_max = max;
        max = c;
      } else if (c > second_max && c != max)
        second_max = c;
    }
  }
  return second_max == 47 ? -1 : second_max - 48;
}

File: Python/Algorithms/1796.py
class Solution:
    def secondHighest(self, s: str) -> int:
        max_, second_max = '', ''
        for i in filter(str.isdigit, s):
            if i > max_:
                if max_ > second_max:
                    second_max = max_
                max_ = i
            elif i > second_max and i != max_:
                second_max = i
        return int(second_max) if second_max else -1
